Amazon.IonObjectMapper/IonSerializer.cs

```diff
@@ -504,5 +533,33 @@ private IIonSerializer GetPrimitiveSerializer<T>(IIonSerializer defaultSerialize
 
             return defaultSerializer;
         }
+
+        private IIonSerializer CreateCustomSerializer (object item)
+        {
+            var customSerializerAttribute = item.GetType().GetCustomAttribute<IonSerializerAttribute>();
+            if (customSerializerAttribute.Factory != null) {
+                var customSerializerFactory = (IIonSerializerFactory)Activator.CreateInstance(customSerializerAttribute.Factory);
+                var customSerializer = customSerializerFactory.Create(options, options.CustomContext);
+                return customSerializer;
+            } else if (customSerializerAttribute.Serializer != null) {
+                // Create customSerializer with serializer attribute

```

Comment:
Should there be something here?

Reply:
We should create customSerializer with IonSerializerAttribute.Serializer attribute here - we currently create customSerializer using IonSerializerAttribute.Factory attribute.

Reply:
Kyle is right, you need to add this case in.

Reply:
Added.